CMakeLists.txt

```diff
@@ -6,6 +6,21 @@ endif()
 
 project("hdr_histogram")
 
+# Follow all steps below in order to calculate new ABI version when updating the library
+# NOTE: THIS IS UNRELATED to the actual project version
+#
+# 1. If the library source code has changed at all since the last update, then increment revision
+# 2. If any interfaces have been added, removed, or changed since the last update, increment current and set revision to 0.
+# 3. If any interfaces have been added since the last public release, then increment age.
+# 4. If any interfaces have been removed since the last public release, then set age to 0.
+
+set(HDR_SOVERSION_CURRENT   1)
+set(HDR_SOVERSION_REVISION  0)
+set(HDR_SOVERSION_AGE       0)
+
+set(HDR_VERSION ${HDR_SOVERSION_CURRENT}.${HDR_SOVERSION_AGE}.${HDR_SOVERSION_REVISION})
```

Comment:
@remicollet Is this order correct? It differs from the explanation in https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html (revision and age are swapped).

Reply:
Yes, I think so

- incr revison will create 1.0.1 (no change)
- incr age will create 1.1.0 (addition)
- incr current will create 2.0.0 (breaking change)

Reply:
The recommendation on that link does use [current]:[revision]:[age], so that if that we are starting with 1.1.1:
* No interfaces changes, increment revision, will give 1.2.1
* Adding interfaces, will give 2.0.2
* Removing/Changing interfaces will give 2.0.0

> If the library source code has changed at all since the last update, then increment revision (‘c:r:a’ becomes ‘c:r+1:a’).

Reply:
The instructions in the comment appear to be consistent with the external libtool documentation, just the order of revision and age is reversed.

The example given by Mike matches my understanding of the instructions. The example from Remi does not, because the addition also needs to touch "current", otherwise repeated additions violate the:

> Also note that age must be less than or equal to the current interface number. 

rule.

------------

Further research leads to: https://www.gnu.org/software/libtool/manual/html_node/Libtool-versioning.html. This means that `current - age` is the lowest supported version, which based on other sources appears to be used on the BSDs at least to generate the filename for the SO file. 

And I'm seeing it's already fixed in e33c19c911bd259a84dcb2b86c2e4bd5b79258e1. I wonder if we should "reset" the so version once to 7.0.0 to ensure there is no confusion going forward?

Reply:
I thought about updating the version, but given there were no actual changes to the "code", i.e. it is still compatible with all uses out there, then I would stick with the same value.